v.5.1.0: Sending reports from Application.attachBaseContext causes a crash

Since 5.1.0, ACRA uses `JobIntentService` for `SenderService`. However, `JobIntentService.CompatWorkEnqueuer` (for pre-Oreo devices) constructor does this:
```
mContext = context.getApplicationContext();
```
as result, `mContext` is `null`, so everything else fails. This is probably an issue with the support library itself, but for ACRA, this means that `ACRA.init` can't be called in `Application.attachBaseContext` anymore.

Calling `ACRA.init` from `Application.onCreate` instead makes everything work again.

> SenderService should not be called in init anyway.
It does when there are unsent crash reports. Here's the full stacktrace:
```
ACRA caught a RuntimeException for com.ballasttechnologies.android.triton
java.lang.RuntimeException: Unable to instantiate application com.ballasttechnologies.android.triton.core.MainApp: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.ComponentName android.content.Context.startService(android.content.Intent)' on a null object reference
    at android.app.LoadedApk.makeApplication(LoadedApk.java:802)
    at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5335)
    at android.app.ActivityThread.-wrap2(ActivityThread.java)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1528)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6077)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
 Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.ComponentName android.content.Context.startService(android.content.Intent)' on a null object reference
    at android.support.v4.app.JobIntentService$CompatWorkEnqueuer.enqueueWork(JobIntentService.java:181)
    at android.support.v4.app.JobIntentService.enqueueWork(JobIntentService.java:522)
    at android.support.v4.app.JobIntentService.enqueueWork(JobIntentService.java:500)
    at org.acra.sender.SenderServiceStarter.startService(SenderServiceStarter.java:54)
    at org.acra.util.ApplicationStartupProcessor.sendApprovedReports(ApplicationStartupProcessor.java:106)
    at org.acra.util.ApplicationStartupProcessor.checkReports(ApplicationStartupProcessor.java:60)
    at org.acra.ACRA.init(ACRA.java:234)
    at org.acra.ACRA.init(ACRA.java:160)
    at com.ballasttechnologies.android.triton.log.CrashReportingConfigurator.execute(CrashReportingConfigurator.java:42)
    at com.ballasttechnologies.android.triton.core.MainApp.initAcra(MainApp.java:63)
    at com.ballasttechnologies.android.triton.core.MainApp.attachBaseContext(MainApp.java:51)
    at android.app.Application.attach(Application.java:189)
    at android.app.Instrumentation.newApplication(Instrumentation.java:1008)
    at android.app.Instrumentation.newApplication(Instrumentation.java:992)
    at android.app.LoadedApk.makeApplication(LoadedApk.java:796)
    at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5335) 
    at android.app.ActivityThread.-wrap2(ActivityThread.java) 
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1528) 
    at android.os.Handler.dispatchMessage(Handler.java:102) 
    at android.os.Looper.loop(Looper.java:154) 
    at android.app.ActivityThread.main(ActivityThread.java:6077) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756) 
```

--- a/acra-core/src/main/AndroidManifest.xml
+++ b/acra-core/src/main/AndroidManifest.xml
@@ -16,6 +16,7 @@
 
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.acra">
+    <uses-permission android:name="android.permission.WAKE_LOCK" android:maxSdkVersion="25"/>
     <application>
         <service
             android:name="org.acra.sender.SenderService"

--- a/acra-core/src/main/java/org/acra/util/ApplicationStartupProcessor.java
+++ b/acra-core/src/main/java/org/acra/util/ApplicationStartupProcessor.java
@@ -21,7 +21,6 @@ import android.content.SharedPreferences;
 import android.content.pm.PackageInfo;
 import android.os.Handler;
 import android.support.annotation.NonNull;
-
 import org.acra.ACRA;
 import org.acra.config.CoreConfiguration;
 import org.acra.file.BulkReportDeleter;
@@ -49,17 +48,20 @@ public final class ApplicationStartupProcessor {
         reportLocator = new ReportLocator(context);
     }
 
-    public void checkReports(boolean enableAcra){
-        if (config.deleteOldUnsentReportsOnApplicationStart()) {
-            deleteUnsentReportsFromOldAppVersion();
-        }
-        if (config.deleteUnapprovedReportsOnApplicationStart()) {
-            reportDeleter.deleteReports(false, 1);
-        }
-        if (enableAcra) {
-            sendApprovedReports();
-            approveOneReport();
-        }
+    public void checkReports(boolean enableAcra) {
+        //application is not ready in onAttachBaseContext, so delay this
+        new Handler(context.getMainLooper()).post(() -> {
+            if (config.deleteOldUnsentReportsOnApplicationStart()) {
+                deleteUnsentReportsFromOldAppVersion();
+            }
+            if (config.deleteUnapprovedReportsOnApplicationStart()) {
+                reportDeleter.deleteReports(false, 1);
+            }
+            if (enableAcra) {
+                sendApprovedReports();
+                approveOneReport();
+            }
+        });
     }
 
     private void approveOneReport() {
@@ -68,10 +70,8 @@ public final class ApplicationStartupProcessor {
         if (reports.length == 0) {
             return; // There are no unapproved reports, so bail now.
         }
-        //cannot directly create stuff from onAttachBaseContext, so defer the call
-        new Handler(context.getMainLooper()).post(() ->
-                //only approve one report at a time to prevent overwhelming users
-                new ReportInteractionExecutor(context, config).performInteractions(reports[0]));
+        //only approve one report at a time to prevent overwhelming users
+        new ReportInteractionExecutor(context, config).performInteractions(reports[0]);
     }
 
     /**

